### PR TITLE
feat: simulate etcd state

### PR DESCRIPTION
**Pull request by Unix4ever**
Implement etcd members, leave, remove member by id.

Add limited support for `List` api, only to fake that etcd directory exists when etcd service is running.

Generate mount statuses out of the machine config.

Make `MachineStatus` controller wait for `apid` and `etcd` services to be up when reporting ready status.

Start `etcd` mock only after we get bootstrap call for the cluster.

Also add gRPC proxy code to the machine simulation. The code is mostly coped intact from Talos, only replaced unix socket with inmem one and dropped auth.

### CODE REVIEW

**Review by smira — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -136,11 +136,9 @@ FROM talemu-linux-${TARGETARCH} AS talemu
 FROM scratch AS talemu-all
 COPY --from=talemu-linux-amd64 / /
 
-FROM scratch AS image-talemu
+FROM alpine AS image-talemu
```
Comment (frezbo) [inline Dockerfile:139]: intended, then needs to go to kres

Comment (Unix4ever) [reply Dockerfile:139]: nope, good catch, forgot to revert

**Review by frezbo — APPROVED** [review]

**Pull request merged**